crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -1410,6 +1410,26 @@ impl VisitAstPath for Analyzer<'_> {
         }
     }
 
+    fn visit_for_of_stmt<'ast: 'r, 'r>(
+        &mut self,
+        n: &'ast ForOfStmt,
+        ast_path: &mut swc_core::ecma::visit::AstNodePath<'r>,
+    ) {
+        let array = self.eval_context.eval(&n.right);
+
```

Comment:
We need to visit `n.right` too. There might be more complex code in that expression